fairworkflows/rdf_wrapper.py

```diff
@@ -48,7 +53,20 @@ def set_attribute(self, predicate, value):
         it (but throw a warning).
         """
         if self.get_attribute(predicate) is not None:
-            warnings.warn(f'A predicate {predicate} was already defined'
-                          f'overwriting {predicate} for {self.self_ref}')
+            warnings.warn(f'A predicate {predicate} was already defined\n'
+                          f'Overwriting {predicate} for {self.self_ref}')
             self._rdf.remove((self.self_ref, predicate, None))
         self._rdf.add((self.self_ref, predicate, value))
+        self._is_modified = True
+
+    def anonymise_rdf(self):
```

Comment:
I am thinking: after publishing `self.self_ref` actually becomes a uri right? So in that case this is not actually replacing the uri with blank nodes. Or am I wrong?

Reply:
```self.self_ref``` is always a blank node. It is now a property of ```RdfWrapper```, and shouldn't be changed by anything under normal circumstances. Publishing changes the value of ```self._uri```, but does not change ```self.self_ref```. When publishing, the self_ref blank node is replaced with whatever the appropriate URI is for that publication, but in the ```FairStep``` or ```FairWorkflow``` object the rdf graph must always be in terms of the same blank node.

Reply:
You're completely right! Indeed self_ref stays self_ref no matter what happens. That makes sense!